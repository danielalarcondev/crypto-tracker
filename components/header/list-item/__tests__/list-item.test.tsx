import * as React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from '@jest/globals';

import ListItem from '../list-item';

const MOCKED_PATHNAME = 'https://test.com/assets';

jest.mock('next/navigation', () => ({
    usePathname: () => MOCKED_PATHNAME
}));

describe('List Item: ', () => {

    it('should render correct data', () => {

        const text = 'Tracks';
        const path = 'https://test.com/tracks';

        const { container } = render(<ListItem text={text} path={path} isMenuOpen={false} />);

        expect(container).toMatchSnapshot();
        expect(container.querySelector('a')?.textContent).toBe(text);
        expect(container.querySelector('a')?.getAttribute('href')).toBe(path);

    });

    it('should have active status', () => {
        const text = 'Assets';
        const path = MOCKED_PATHNAME;

        const { container } = render(<ListItem text={text} path={path} isMenuOpen={false} />);
		
        expect(container).toMatchSnapshot();
        expect(container.querySelector('li')?.className).toContain('border-primary-500');
    });

    it('should not have active status', () => {
        const text = 'Tracker';
        const path = 'https://test.com/tracker';

        const { container } = render(<ListItem text={text} path={path} isMenuOpen={false} />);
        expect(container.querySelector('li')?.className).not.toContain('border-primary-500');
    });
});
