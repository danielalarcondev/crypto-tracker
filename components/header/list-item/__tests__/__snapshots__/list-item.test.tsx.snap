// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item:  should have active status 1`] = `
<div>
  <li
    class="pl-3 pr-3 text-xl hover:cursor-pointer border-b-4 hover:text-primary-500 border-primary-500 w-fit h-[calc(100%+1px)]"
  >
    <a
      class="menu-anchor block md:bg-transparent h-full flex items-center mt-1"
      href="https://test.com/assets"
    >
      Assets
    </a>
  </li>
</div>
`;

exports[`List Item:  should render correct data 1`] = `
<div>
  <li
    class="pl-3 pr-3 text-xl hover:cursor-pointer border-b-4 hover:text-primary-500 border-transparent w-fit h-[calc(100%+1px)]"
  >
    <a
      class="menu-anchor block md:bg-transparent h-full flex items-center mt-1"
      href="https://test.com/tracks"
    >
      Tracks
    </a>
  </li>
</div>
`;
